haproxy-operator/templates/haproxy.cfg.j2

```diff
@@ -1,3 +1,12 @@
+{%- macro ddos_protection_rules() -%}
+{% if not disable_ddos_protection %}
```

Comment:
I can't foresee any scenario where this will need to be disabled, except for, perhaps, the header host when testing things. But tweaking the request to comply for a testing scenario seems way easier that changing a charm config.
I'd remove the option altogether